### PR TITLE
VIA changes for 7.1.x

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
This implements the VIA protocol stack support currently in master along with the necessary infrastructure changes. It is split in to 3 commits

* Add `MemView` / `StringView`.
* Change the infrastructure of the protocol stack logic to use `StringView`.
* Changes the VIA header logic and add documentation.

The VIA tests unfortunately were not backported because they depend on significant changes to the testing infrastructure.

### CODE REVIEW

**Comment by SolidWallOfCode** [comment]
[approve ci autest]

**Comment by SolidWallOfCode** [comment]
[approve ci autest]

**Comment by SolidWallOfCode** [comment]
[approve ci autest]

File: doc/admin-guide/files/records.config.en.rst [hunk 1]
```diff
@@ -820,18 +820,19 @@ ip-resolve
...
+   ===== ====================================================================
    Value Effect
-   ===== ============================================
+   ===== ====================================================================
    ``0`` Do not modify or set this Via header.
-   ``1`` Update the Via, with normal verbosity.
-   ``2`` Update the Via, with higher verbosity.
-   ``3`` Update the Via, with highest verbosity.
-   ===== ============================================
+   ``1`` Add the basic protocol and proxy identifier.
```
Comment (SolidWallOfCode) [inline doc/admin-guide/files/records.config.en.rst]: Isn't that the change in the docs?

**Comment by zwoop** [comment]
As unfortunate as it is, we have to fix this one way or another, because the changes made to 7.0.0 breaks the RFC for the Via header (which fucked us up).

**Pull request merged**